astropy/time/core.py

```diff
@@ -2313,7 +2313,16 @@ def __init__(self, val, val2=None, format=None, scale=None, copy=False):
                 self._set_scale(scale)
         else:
             if format is None:
-                format = 'datetime' if isinstance(val, timedelta) else 'jd'
+                if isinstance(val, timedelta):
+                    format = 'datetime'
+                else:
+                    if getattr(val, 'unit', None) is None:
+                        warn(
+                            "Numerical value without unit passed to TimeDelta"
+                            ", assuming days",
+                            AstropyDeprecationWarning
```

Comment:
I'd like this to be a subclass. In my `Time` subclass I'll include a line to ignore this warning, but I don't want to ignore `AstropyDeprecationWarning` globally, just this particular warning.

Reply:
Done